projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1001,15 +1042,37 @@ export class IgxTimePickerComponent implements
     private _convertMinMaxValue(value: string): Date {
         const date = this.value ? new Date(this.value) : this._dateFromModel ? new Date(this._dateFromModel) : new Date();
         const sections = value.split(/[\s:]+/);
+        let hour, minutes, amPM;
 
-        date.setHours(parseInt(sections[0], 10));
-        date.setMinutes(parseInt(sections[1], 10));
         date.setSeconds(0);
-        if (sections[2] && sections[2] === 'PM' && sections[0] !== '12') {
-            date.setHours(date.getHours() + 12);
+
+        if (this.showHoursList) {
+            hour = sections[0];
+            date.setHours(parseInt(hour, 10));
         }
-        if (sections[0] === '12' && sections[2] && sections[2] === 'AM') {
-            date.setHours(0);
+
+        if (this.showMinutesList) {
+            minutes = this.showHoursList ? sections[1] : sections[0];
+            date.setMinutes(parseInt(minutes, 10));
+        }
+
+        if (this.showAmPmList) {
+            amPM = sections[sections.length - 1];
+
+            if (this.showHoursList && amPM === 'PM' && date.getHours().toString() !== '12') {
+                date.setHours(date.getHours() + 12);
+            }
+
+            if (!this.showHoursList && amPM === 'PM' && date.getHours().toString() <= '11') {
```

Comment:
merge with the above condition to avoid duplicating code (date.setHours(date.getHours() + 12))

Reply:
I'd rather not, this logic is very fragile.

Reply:
You could merge those with || and the result will be the same.